<launch>
  <arg name="port" default="/dev/ttyACM0" />

  <!-- cases where this being required would not be desirable? optional through arg? -->
  <node name="arduino_interface" pkg="rosserial_python" type="serial_node.py" output="screen" required="true">
    <!-- TODO figure out how to autodetect and use hw id / software to identify by function -->
    <param name="port" value="$(arg port)" type="str" />
    <param name="baud" value="9600" />
    <param name="sync_timeout" value="False" type="bool" />

    <!-- TODO test it fails / how if not advertised -->
    <rosparam>
      require:
        publishers: []
        subscribers: []
        services: ['load_defaults', 'load_seq']
    </rosparam>
  </node>
</launch>
